lib/ruby_lsp/requests/support/rubocop_runner.rb

```diff
@@ -76,6 +78,11 @@ def formatted_source
         def file_finished(_file, offenses)
           @offenses = offenses
         end
+
+        sig { returns(T::Boolean) }
+        def rubocop_version_supports_raise_cop_error?
+          Gem::Version.new(@rubocop_version) >= Gem::Version.new("1.38.0")
```

Comment:
I don't think we should be doing version checks. Can't we do feature detection? We have access to the whole `RuboCop` namespace, so we can just check for the existence of `RuboCop::ErrorWithAnalyzedFileLocation`

Reply:
`RuboCop::ErrorWithAnalyzedFileLocation` already existed prior to the flag, but I'll see if there's another way.

Reply:
Ah right, sorry I read the PR wrong.

This seems to work for doing feature detection, btw:
```ruby
irb> RuboCop::Options.new.parse(["--foo", "bar"])
OptionParser::InvalidOption: invalid option: --foo
```

Reply:
Confirmed that this works:
```ruby
[1] pry> gem("rubocop", "1.36")
=> true
[2] pry(main)> require "rubocop"
=> true
[3] pry(main)> RuboCop::Options.new.parse(["--raise-cop-error"])
OptionParser::InvalidOption: invalid option: --raise-cop-error
from /Users/ufuk/.gem/ruby/3.1.2/gems/rubocop-1.36.0/lib/rubocop/options.rb:28:in `parse'
```
vs
```ruby
[1] pry(main)> gem("rubocop", "1.39")
=> true
[2] pry(main)> require "rubocop"
=> true
[3] pry(main)> RuboCop::Options.new.parse(["--raise-cop-error"])
=> [{:raise_cop_error=>true}, []]
```